pyo3-macros-backend/src/method.rs

```diff
@@ -63,6 +63,10 @@ impl<'a> FnArg<'a> {
             }
         }
     }
+
+    pub fn is_regular(&self) -> bool {
+        !self.py && !self.is_cancel_handle && !self.is_kwargs && !self.is_varargs
+    }
```

Comment:
Seeing this I'm thinking `FnArg` might want to be reworked to be an `enum`. Not sure if it's worth trying to make that part of this PR or we punt for another time.